### ISSUE TITLE
Footgun in the Makefile when Limine fails to build

### ISSUE THREAD

**Issue by sagehane** [48cf/limine-zig-template#8]
### Issue

If Limine fails to build with `make limine`, subsequent attempts to build it will silently do nothing.
This can be easily triggered on systems like NixOS or Guix in which a person might have an environment with no C compilers in `$PATH`.

It causes all other steps depending on Limine to fail until performing `make distclean` and rebuilding properly.

### How to reproduce

1. Fail to build Limine (reproduced on NixOS without any C compiler in the environment)
2. Try to build Limine again

```sh-session
$ make limine
git clone https://github.com/limine-bootloader/limine.git --branch=v7.x-binary --depth=1
Cloning into 'limine'...
remote: Enumerating objects: 18, done.
remote: Counting objects: 100% (18/18), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 18 (delta 5), reused 11 (delta 5), pack-reused 0
Receiving objects: 100% (18/18), 618.16 KiB | 10.84 MiB/s, done.
Resolving deltas: 100% (5/5), done.
make -C limine
make[1]: Entering directory '/home/plumeus/repos/limine-zig-template/limine'
cc -g -O2 -pipe -Wall -Wextra    -std=c99 limine.c  -o limine
make[1]: cc: No such file or directory
make[1]: *** [Makefile:38: limine] Error 127
make[1]: Leaving directory '/home/plumeus/repos/limine-zig-template/limine'
make: *** [GNUmakefile:47: limine] Error 2

$ make limine
make: 'limine' is up to date.
```

**Comment by 48cf** [comment]
It seems to be a problem with every other Limine template as the makefile is mostly shared between all of them. I'm surprised it was never brought up by anyone, considering that Zig isn't the most popular language out there. I am not sure how to properly handle this so I'll keep this issue open for the time being in case anyone else hits that.

**Comment by Mintsuki** [comment]
Fixed by c972157, thanks
